WalletWasabi.Tests/Helpers/ArenaBuilder.cs

```diff
@@ -0,0 +1,70 @@
+using Moq;
+using NBitcoin;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.BitcoinCore.Rpc;
+using WalletWasabi.WabiSabi.Backend;
+using WalletWasabi.WabiSabi.Backend.Banning;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+
+namespace WalletWasabi.Tests.Helpers
+{
+	/// <summary>
+	/// Builder class for <see cref="Arena"/>.
+	/// </summary>
+	public class ArenaBuilder
+	{
+		public TimeSpan? Period { get; set; }
+		public Network? Network { get; set; }
+		public WabiSabiConfig? Config { get; set; }
+		public IRPCClient? Rpc { get; set; }
+		public Prison? Prison { get; set; }
+
+		/// <param name="rounds">Rounds to initialize <see cref="Arena"/> with.</param>
+		public Arena Create(params Round[] rounds)
+		{
+			TimeSpan period = Period ?? TimeSpan.FromHours(1);
+			Prison prison = Prison ?? new();
+			WabiSabiConfig config = Config ?? new();
+			IRPCClient rpc = Rpc ?? WabiSabiFactory.CreatePreconfiguredRpcClient().Object;
+			Network network = Network ?? Network.Main;
+
+			Arena arena = new(period, network, config, rpc, prison);
+
+			foreach (var round in rounds)
+			{
+				arena.Rounds.Add(round);
+			}
+
+			return arena;
+		}
+
+		public Task<Arena> CreateAndStartAsync(params Round[] rounds)
+			=> CreateAndStartAsync(rounds, CancellationToken.None);
+
+		public async Task<Arena> CreateAndStartAsync(Round[] rounds, CancellationToken cancellationToken = default)
+		{
+			Arena? toDispose = null;
+
+			try
+			{
+				toDispose = Create(rounds);
+				Arena arena = toDispose;
+				await arena.StartAsync(cancellationToken).ConfigureAwait(false);
+				toDispose = null;
+				return arena;
+			}
+			finally
+			{
+				toDispose?.Dispose();
+			}
```

Comment:
So this method was implemented as it is now to follow CA2000 [recommended approach](https://docs.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca2000#example-1) to disposing an object (`arena`).

`catch { arena?.Dispose(); throw; }` seems to work too but `finally` seems cleaner to me as it follows the recommendation.

It may be perceived as an overkill at this point but if we want to write code that has a chance to be long-lived this is probably better.

Reply:
But yes, your code with `catch` would be probably shorter and maybe mentally easier to grasp. I got used to the solution with `finally` so I use it somewhat automatically.

The thing I don't really know is whether `catch { arena?.Dispose(); throw; }` and `finally` differs in performace (not that it is important here) because of that re-throw. So I can't say whether `finally` and `catch { arena?.Dispose(); throw; }` behave identically we would need to put it into ILSpy or sharplab to see produced bytecode.